src/models/pie.js

```diff
@@ -338,6 +338,38 @@ nv.models.pie = function() {
                         }
                         return label;
                     })
+                    .each(function (d, i) {
+                        if (!this.getBBox) return;
+                        var bb = this.getBBox(),
+                        center = labelsArc[i].centroid(d);
+                        var topLeft = {
+                          x : center[0] + bb.x,
+                          y : center[1] + bb.y
+                        };
+
+                        var topRight = {
+                          x : topLeft.x + bb.width,
+                          y : topLeft.y
+                        };
+
+                        var bottomLeft = {
+                          x : topLeft.x,
+                          y : topLeft.y + bb.height
+                        };
+
+                        var bottomRight = {
+                          x : topLeft.x + bb.width,
+                          y : topLeft.y + bb.height
+                        };
+
+                        d.visible = nv.utils.pointIsInArc(topLeft, d, arc) &&
+                        nv.utils.pointIsInArc(topRight, d, arc) &&
+                        nv.utils.pointIsInArc(bottomLeft, d, arc) &&
+                        nv.utils.pointIsInArc(bottomRight, d, arc);
+                    })
+                    .style('display', function (d) {
+                        return d.visible ? null : 'none';
+                    })
```

Comment:
I guess I'd be open to merging it if it was controlled by an option and off by default (for backward compatibility).   I can just see other people not wanting this and considering it a bug.